src/PostgREST/ApiRequest.hs

```diff
@@ -132,6 +136,14 @@ userApiRequest schema req reqBody
       , iCookies = fromMaybe [] $ parseCookiesText <$> lookupHeader "Cookie"
       }
  where
+  (filters, rpcQParams) =
+    case action of
+      ActionInvoke{isReadOnly=True} -> (filter (liftM2 (||) (isEmbedPath . fst) (hasOperator . snd)) flts,
```

Comment:
Perhaps the [partition](https://hackage.haskell.org/package/base-4.10.0.0/docs/Data-List.html#v:partition) function could clean up the code here where you're splitting a list into two which do and do not satisfy a predicate.